### DISCUSSION TITLE
Consider using parking lot types

### DISCUSSION THREAD

**Discussion by SuchAFuriousDeath** [obhq/obliteration#786] [General]
The crate https://docs.rs/parking_lot/latest/parking_lot/ claims that it implements faster, smaller and more efficient implementations of 
Mutex, RwLock, Condvar and Once than the standard library. Many big Rust crates use it internally.

**Comment by ultimaweapon** [comment]
I'm not a fan of using an alternative to `std` things unless the difference is huge. The reason is because it cause confusion when reading the code like when encountering `RwLock`, which implementation is this? `std` or something else?

**Comment by SuchAFuriousDeath** [comment]
We could use type aliases.

**Reply by ultimaweapon** [reply]
The real issue here is does it make huge differences with `std`?

**Reply by SuchAFuriousDeath** [reply]
I saw a speed comparison somewhere, but I can't find it now :D

**Reply by SoffiMeower** [reply]
https://github.com/Amanieu/parking_lot?tab=readme-ov-file#features

Also at the top of the readme:
"When tested on x86_64 Linux, parking_lot::Mutex was found to be 1.5x faster than std::sync::Mutex when uncontended, and up to 5x faster when contended from multiple threads. The numbers for RwLock vary depending on the number of reader and writer threads, but are almost always faster than the standard library RwLock, and even up to 50x faster in some cases."

@ultimaweapon

**Reply by ultimaweapon** [reply]
Let's use `std` version until we have some performance problem then we can try `parking_lot`.
